table/partitioned_filter_block_test.cc

```diff
@@ -66,12 +66,13 @@ class PartitionedFilterBlockTest : public testing::Test {
   }
 
   uint64_t MaxFilterSize() {
+    uint32_t dont_care1, dont_care2;
```

Comment:
> An assert will break the build
You mean compile time issue?!
If you mean that the test will fail, that is the intended behavior to catch the unexpected issue.  

> static_cast will almost certainly fail the test too 
"almost certainly" means that we are leaving our fate on being lucky. 

> It will get caught or not caught either way
Moreover imagine a new developer in five years. When she sees the assert failure she can quickly tell what was the expected behavior and what is the issue. Otherwise she would have to spend much time in debugging a unit test that she does not know. Developer are not expected to learn all unit tests and a good unit test should be easy to understand and point out the problem quickly without wasting much development time.

> For release build, the assert is removed
This is a unit test. Who would run the unit test with a release build?

> I think this cast is a legacy problem
You have a good habit of giving references for your views. Can you kindly provide one for this too?